gaphor/SysML/diagramframe.py

```diff
@@ -0,0 +1,161 @@
+from gaphas.geometry import Rectangle
+
+from gaphor.core.modeling import DrawContext, UpdateContext
+from gaphor.core.styling import (
+    FontWeight,
+    TextAlign,
+)
+from gaphor.diagram.deletable import item_explicitly_deletable
+from gaphor.diagram.diagramlabel import diagram_label
+from gaphor.diagram.presentation import (
+    Classified,
+    ElementPresentation,
+    connect,
+)
+from gaphor.diagram.shapes import (
+    cairo_state,
+)
+from gaphor.diagram.text import Layout
+from gaphor.UML.actions.activity import ActivityParameterNodeItem
+from gaphor.UML.uml import Activity, ActivityParameterNode
+
+
+class DiagramFrameItem(Classified, ElementPresentation):
+    def __init__(self, diagram, id=None):
+        super().__init__(diagram, id)
+
+        self.watch("diagram[Diagram].name", self.update_shapes).watch(
+            "diagram[Diagram].diagramType", self.update_shapes
+        ).watch("diagram[Diagram].element", self.update_element).watch(
+            "subject", self.update_element
+        ).watch("subject[Activity].node", self.update_frame_attachments).watch(
+            "diagram[Diagram].element[NamedElement].name", self.update_shapes
+        )
+
+    def update_element(self, event=None):
+        if not self.diagram or not self.diagram.element:
+            return
+
+        self.subject = self.diagram.element
+        self.update_shapes()
+        self.update_frame_attachments()
+
+    def postload(self):
+        super().postload()
+        self.update_frame_attachments()
+
+    def update_shapes(self, event=None):
+        if not self.diagram:
+            return
+
+        self.shape = DiagramFrameShape(
+            kind=self.diagram.diagramType,
+            label=diagram_label(self.diagram),
+        )
+
+    def update_frame_attachments(self, event=None):
+        if not self.subject:
+            return
+
+        if isinstance(self.subject, Activity):
+            self.update_activity_parameters()
+
+    def update_activity_parameters(self):
```

Comment:
Now it's only activity diagrams. I guess you have an idea of how to extend this to other diagram types?